marketplace.js

```diff
@@ -0,0 +1,60 @@
+
+function Cart(){
+	
+    this.itemGroups = [];
+	this.items = 0;
+	this.totalPrice = 0;
+	
+	this.getTotalAmount = function(){
+		
+			 this.items = 0;
+			 this.totalPrice = 0;
+			
+			for(let i = 0; i <this.itemGroups.length; i++){
+				let currGroup = this.itemGroups[i];
+				this.items = this.items + currGroup.numberOfItems;
+				this.totalPrice = this.totalPrice + (currGroup.numberOfItems * currGroup.pricePerItem);
+			}
+			return 1;
+	}
+	
+    this.showTotalAmount = function(){
+        if (this.itemGroups.length == 0){
+            document.write("<p> You have 0 item, for a total amount of 0$, in your cart! </p>");
+        } 
+		else  {
+			this.getTotalAmount();
```

Comment:
This works well, but is not typically a good usage of "this". Instead, we would usually just return the value directly (instead of returning 1).